base/src/main/java/io/spine/base/Environment.java

```diff
@@ -32,17 +32,82 @@
 /**
  * Provides information about the environment (current platform used, etc.).
  *
- * <p>Allows to determine the current environment type. {@linkplain BaseEnvironmentType
- * 2 environment types} exist by default.
+ * <h1>Environment Type Detection</h1>
  *
- * <p>{@code Environment} is a singleton.
+ * <p>Current implementation allows to detect the type of the current environment. The framework
+ * brings two environment types out-of-the-box:
+ *
+ * <ul>
+ *     <li><em>{@link BaseEnvironmentType#TESTS TESTS}</em> is detected
+ *     if the current call stack has a reference to the unit testing framework.
+ *
+ *     <li><em>{@link BaseEnvironmentType#PRODUCTION PRODUCTION}</em> is set in all other cases.
+ * </ul>
+ *
+ * <p>The framework users may define their custom settings depending on the {@linkplain}current
+ * environment type:
+ *
+ * <pre>
+ *
+ * public final class Application {
+ *
+ *     private final EmailSender sender;
+ *
+ *     private Application() {
+ *         EnvironmentType type = Environment.instance()
+ *                                           .type();
+ *         if(type == BaseEnvironmentType.TESTS) {
+ *             // Do not send out emails if in tests.
+ *             this.sender = new MockEmailSender();
+ *         } else {
+ *             this.sender = EmailSender.withConfig("email_gateway.yml");
+ *         }
+ *         //...
+ *     }
+ * }
+ * </pre>
  *
  * <h1>Custom environment types</h1>
- * {@code Environment} allows to {@link #register(EnvironmentType) reguster custom types}. If done,
- * {@code Environment} can then check whether the specified type is currently enabled.
+ *
+ * {@code Environment} allows to {@link #register(EnvironmentType) reguster custom types}.
+ * In this case the environment detection functionality iterates over all known types, starting
+ * with those registered by the framework user:
+ *
+ * <pre>
+ *
+ * public final class Application {
+ *
+ *     static {
+ *         Environment.instance()
+ *                    .register(StagingEnvironmentType.instance())
+ *                    .register(LoadTestingType.instance());
+ *     }
+ *
+ *     private final ConnectionPool pool;
+ *
+ *     private Application() {
+ *         EnvironmentType type = Environment.instance()
+ *                                           .type();
+ *         if (type == BaseEnvironmentType.TESTS) {
+ *              // Single connection is enough for tests.
+ *             this.pool = new ConnectionPoolImpl(PoolCapacity.of(1));
+ *         } else {
+ *             if(LoadTestingType.instance().enabled()) {
+ *                 this.pool =
```

Comment:
Is this what we would expect people to do?

I thought that 
```java
else {
    if(type == LoadTestingType.instance()) {
       ...
    }
}
```
was the intended way to check the environment.